Update Android to Current Square Documentation

zacdemi opened the pull request:
Fixes #7 

## Context
Android configuration files out of date with square docs. 

## Solution
update build.gradle and manifest files
https://developer.squareup.com/docs/pos-api/build-on-android

zacdemi committed:
update build.gradle for > Gradle 3.0.0 and later

zacdemi committed:
update for Android 11 or newer

zacdemi committed:
minsdk to 16

zacdemi committed:
update to latest ios sdk

zacdemi committed:
gift card support ios

xfyre commented:
I can confirm that this PR fixes the problem with the Square app. Can we merge & release it?

zacdemi committed:
Update for React 72

-Replace depreciated DeviceEventEmitter with NativeEventEmitter (Android)
-Replace depreciated removeEventListener with remove (iOS)
